### PR TITLE
fix alignLeftAt: alignRightAt: to better fit the implementation of Vertical Alignment

### PR DESCRIPTION
**Pull request by Enzo-Demeulenaere**
I spotted this issue while in a call with @tinchodias, where in the BlClickEventExamplesTest>>threeElementsWithFalseChildClipping, the placement of elements would not feel right when reading alignRightAt: and alignTopAt: .

I noticed that Vertical Alignment used "BlVerticalCoordinate**Top**Alignment" and "BlVerticalCoordinate**Bottom**Alignment" while Horizontal Alignment used only "BlHorizontalCoordinate**Start**Alignment" (while also not having any method alignLeftAt: when dealing with ignored layout)

I propose this fix that matches the implementation of Vertical alignment, describing my methods as such :

alignLeftAt: Align the left of the element horizontally with a unit-coordinate inside of the parent, going from left to right

alignRightAt: "Align the right of the element horizontally with a unit-coordinate inside of the parent, going from right to left"

### CODE REVIEW

**Review by tinchodias — COMMENTED** [review]
@Enzo-Demeulenaere thanks for the change.

Maybe you can add some test? I didn't check where should it be. I guess in the frame layout tests, and ignore layout tests?

What do you think?

File: src/Bloc-Layout/BlFrameLayoutConstraintsHorizontal.class.st [hunk 1]
```diff
@@ -33,7 +33,7 @@ BlFrameLayoutConstraintsHorizontal >> alignLeft [
 BlFrameLayoutConstraintsHorizontal >> alignLeftAt: aUnitCoordinateNumber [
 	"Align the left of the element horizontally with a unit-coordinate inside of the parent"
 
-	alignment := BlHorizontalCoordinateStartAlignment new
+	alignment := BlHorizontalCoordinateLeftAlignment new
 		coordinate: aUnitCoordinateNumber;
 		relativeAlignment: BlElementAlignment horizontal start
```
[the remark this answers is not in the document]
Comment (Enzo-Demeulenaere) [reply src/Bloc-Layout/BlFrameLayoutConstraintsHorizontal.class.st:38]: Hi, 

By looking at the code I would say this doesn't change anything as the coordinate: and relativeAlignment: methods are defined and called in the 'BlHorizontalCoordinateAlignment' superclass

File: src/Bloc-Layout/BlFrameLayoutConstraintsHorizontal.class.st [hunk 2]
```diff
@@ -52,7 +52,7 @@ BlFrameLayoutConstraintsHorizontal >> alignRight [
 BlFrameLayoutConstraintsHorizontal >> alignRightAt: aUnitCoordinateNumber [
 	"Align the right of the element horizontally with a unit-coordinate inside of the parent"
 
-	alignment := BlHorizontalCoordinateStartAlignment new
+	alignment := BlHorizontalCoordinateRightAlignment new
 		coordinate: aUnitCoordinateNumber;
 		relativeAlignment: BlElementAlignment horizontal end
```
[the remark this answers is not in the document]
Comment (Enzo-Demeulenaere) [reply src/Bloc-Layout/BlFrameLayoutConstraintsHorizontal.class.st:57]: I see what you mean but now that we have a `RightAlignment` with an `EndAlignment` does this mean it goes from right to left ? or just from right to right somehow ?

[the remark this answers is not in the document]
Comment (tinchodias) [reply src/Bloc-Layout/BlFrameLayoutConstraintsHorizontal.class.st:57]: I guess that if a layout were in rightToLeft mode, then start means right and end means left. But in the most usual leftToRight mode yes, start=left and end=right. But not sure how is it working.

File: src/Bloc-Layout/BlFrameLayoutConstraintsHorizontal.class.st (hunk 1 above)
[the remark this answers is not in the document]
Comment (tinchodias) [reply src/Bloc-Layout/BlFrameLayoutConstraintsHorizontal.class.st:38]: I'm confused with this code (not the change but the whole code). This "aUnitCoordinateNumber" is probably a [0..1] ratio. I find strange the term coordinate, or am I wrong?

**Comment by Enzo-Demeulenaere** [comment]
Hi @Martin I just added tests as well as a small cheat sheet (in Bloc-Examples) for anyone to see it when trying to understand `align[Direction]At:` 

Here's how it looks 
Capture d’écran 2024-01-23 à 10 09 02

**Comment by tinchodias** [comment]
looks great thanks

**Pull request merged**